Controllers/LeaderboardController.cs

```diff
@@ -24,33 +39,100 @@ public IActionResult GetLeaderboard()
         }
 
         [HttpPost("NewEntry")]
-        public IActionResult AddNewEntry([FromBody] Entry entry)
+        public IActionResult NewEntry([FromForm] EditDto entry)
         {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest();
+            }
+
+            if (!_isAdmin)
+            {
+                return Unauthorized();
+            }
+
             _leaderboard.InsertEntry(entry.Name1, entry.Name2, entry.Score);
             _hubContext.Clients.All.SendAsync("Update");
-            return Ok("Entry added");
+            return Redirect(Request.Headers["Referer"]);
```

Comment:
For my personnal understanding: why?

Reply:
It's because when I do a 'Html.BeginForm()' in .cshtml I get redirect to the controller/action in the url. With this I send the user back to where is was (for now it's alway the admin page, but if we want to have another page with those controls it should handle it)

Reply:
We could still Log "Entry added"